airbyte-integrations/connectors/source-metabase/source_metabase/source.py

```diff
@@ -3,117 +3,35 @@
-
-    def get_new_session_token(self, username: str, password: str) -> str:
-        response = requests.post(
-            f"{self.api_url}session", headers={"Content-Type": "application/json"}, json={"username": username, "password": password}
-        )
-        response.raise_for_status()
-        if response.ok:
-            self.session_token = response.json()["id"]
-            self.need_session_close = True
-            self.logger.info(f"New session token generated for {username}")
-        else:
-            raise ConnectionError(f"Failed to retrieve new session token, response code {response.status_code} because {response.reason}")
-        return self.session_token
-
-    def has_valid_token(self) -> bool:
-        try:
-            response = requests.get(f"{self.api_url}user/current", headers=self.get_auth_header())
-            response.raise_for_status()
-        except requests.exceptions.HTTPError as e:
-            if e.response.status_code == 401:
-                self.logger.warn(f"Unable to connect to Metabase source due to {str(e)}, retrying with a new session_token...")
-                self.get_new_session_token(self.username, self.password)
-                response = requests.get(f"{self.api_url}user/current", headers=self.get_auth_header())
-                response.raise_for_status()
-            else:
-                raise ConnectionError(f"Error while checking connection: {e}")
-        if response.ok:
-            json_response = response.json()
-            self.logger.info(
-                f"Connection check for Metabase successful for {json_response['common_name']} login at {json_response['last_login']}"
-            )
-            return True
-        else:
-            raise ConnectionError(f"Failed to retrieve new session token, response code {response.status_code} because {response.reason}")
-
-    def get_auth_header(self) -> Mapping[str, Any]:
-        return {"X-Metabase-Session": self.session_token}
-
-    def close_session(self):
-        if self.need_session_close:
-            response = requests.delete(
-                f"{self.api_url}session", headers=self.get_auth_header(), json={"metabase-session-id": self.session_token}
-            )
-            response.raise_for_status()
-            if response.ok:
-                self.logger.info("Session successfully closed")
-            else:
-                self.logger.info(f"Unable to close session {response.status_code}: {response.reason}")
-        else:
-            self.logger.info("Session was not opened by this connector.")
-
-
-class SourceMetabase(AbstractSource):
+class SourceMetabase(YamlDeclarativeSource):
     def __init__(self):
         self.session = None
+        super().__init__(**{"path_to_yaml": "metabase.yaml"})
 
     def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
         session = None
         try:
-            session = MetabaseAuth(logger, config)
```

Comment:
On one hand you are defining a check in yaml with a stream passed in, on the other hand you're overriding a `check` method in python. This means yaml check will not work and has no sense. We have to leave just one of them, I'm in favour of yaml based check

Reply:
thanks, we need custom check because of updating session token, so I left check here, and removed from yaml.